### PR TITLE
Update obstacle avoidance version 1.0.1

### LINKED ISSUE #16: Sense Obstacles
3.1 As a user, I want my car to automatically avoid obstacles so that I do not have to manually avoid them.

FR 1.1

### LINKED ISSUE #20: Code structure set up
8.1 As a  Developer I want a basic code structure that i can incrementally improve

Accpetance criteria:
Basic set up of directories.
Environment set up. 
Mqtt broker set up,
Basic arduino file to test on

### LINKED ISSUE #22: Refactor SmartCar .ino file
8.2 As a developer I want have a well structured understandable smart-car program that so that I can easily improve and scale the enterprise

### PR DESCRIPTION
**Pull request by drakeaxelrod**
Moved logic for obstacle avoidance from python to Arduino file.
This creates a more reliable and robust safety mechanism. Minor 
refactoring of both the python code and the Arduino file.

### CODE REVIEW

**Comment by drakeaxelrod** [comment]
A lot of merge requests because we didn't know that we had to add the dependencies to the CI.

**Review by MrJonte — APPROVED** [review]
Was a part of all of the coding there for I approve of this code!

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
+    "python.pythonPath": "C:\\Users\\Drake\\.virtualenvs\\group-02-uML8bwF_\\Scripts\\python.exe",
+    // "python.formatting.provider": "black"
+}
```
Comment (platisd) [inline .vscode/settings.json:10]: You probably don't want to version control this file

File: Pipfile.lock [hunk 2]
```diff
@@ -0,0 +1,203 @@
+{
```
Comment (platisd) [inline Pipfile.lock:1]: You probably don't want to version control this file either

File: arduino/smartcar/smartcar.ino [hunk 3]
```diff
@@ -1,9 +1,156 @@
...
+boolean checkSensor(int sensorData, int max)
+{
+    if (0 < sensorData && sensorData < max)
+    {
+        return true;
+    }
+    else
+    {
+        return false;
+    }
```
Comment (platisd) [inline arduino/smartcar/smartcar.ino:137]: What about just `return 0 < sensorData && sensorData < max;`

File: .gitignore [hunk 4]
```diff
@@ -30,3 +30,18 @@
...
+# vsc
+.vscode
+
+# python
+__pycache__
+
+# misc
+test.ino
+local.py
+.car
```
Comment (platisd) [inline .gitignore:47]: I can't leave a comment there, so I am commenting here: You probably should not version control `.pyc` files

**Review by platisd — APPROVED** [review]
The obstacle avoidance part looks good and thus approved. 

There are other issues that I've left comments about, feel free to address them at your own convenience.

File: arduino/smartcar/smartcar.ino (hunk 3 above)
Comment (MrJonte) [reply arduino/smartcar/smartcar.ino:137]: Good comment, one of those things you don't see when you stared at the same code for to long. 
Thank you.

**Pull request merged**